app/controllers/spree/api/products_controller_decorator.rb

```diff
@@ -11,11 +14,16 @@ def managed
 
   # TODO: This should be named 'managed'. Is the action above used? Maybe we should remove it.
   def bulk_products
-    @products = OpenFoodNetwork::Permissions.new(current_api_user).editable_products.
-      merge(product_scope).
-      order('created_at DESC').
+    product_query = OpenFoodNetwork::Permissions.new(current_api_user).
+      editable_products.merge(product_scope)
+
+    if params[:import_date].present?
+      product_query = product_query.imported_on(params[:import_date]).group_by_products_id
+    end
+
+    @products = product_query.order('created_at DESC').
       ransack(params[:q]).result.
-      page(params[:page]).per(params[:per_page])
+      page(params[:page] || DEFAULT_PAGE).per(params[:per_page] || DEFAULT_PER_PAGE)
```

Comment:
I wanted to ask you to investigate the gem's config options so that we ensure `params[:per_page]` it's within a defined upper limit so no malicious users can blow up the server but... I don't even see which gem we're using so can you please specify it in our Gemfile? this is implicit dependency on Spree is particularly scary because I assume we're using whatever they might have defined.

Reply:
Looks like it's `Kaminari`, and yes it's included by Spree.

Reply:
Ok, I've added an entry in the Gemfile (with a note), and added an initializer with a max_per_page value for Kaminari. :+1: